chipsec/modules/tools/uefi/uefivar_fuzz.py

```diff
@@ -169,25 +172,33 @@ def run( self, module_argv ):
             self.logger.log( 'Seed      : %d' % SEED )
             self.logger.log( 'Test case : %d' % CASE )
             self.logger.log('')
-     
             for count in range(1,ITERATIONS+CASE):
                 
                 if FUZZ_NAME:
                     _NAME = ''
-                    for n in range(int(self.rnd(3),16)):
-                        _NAME += random.choice(string.printable)
+                    while not _NAME:
+                        for n in xrange(random.randrange(BOUND_STR)):
+                            _NAME += random.choice(string.printable)    
+
                 if FUZZ_GUID  : _GUID   = self.rnd(4)+'-'+self.rnd(2)+'-'+self.rnd(2)+'-'+self.rnd(2)+'-'+self.rnd(6)
+                
                 if FUZZ_ATTRIB: _ATTRIB = int(self.rnd(4),16)
-                if FUZZ_DATA  : _DATA   = self.rnd(int(self.rnd(3),16))
-                if FUZZ_SIZE  : _SIZE   = int(self.rnd(3),16)
-
+                
+                if FUZZ_DATA  : 
+                    _DATA = None
+                    while not _DATA:
+                        _DATA   = self.rnd(random.randrange(BOUND_INT))
+                        
+                if FUZZ_SIZE  : _SIZE   = int(self.rnd(3),16)            
```

Comment:
Do we want the buffer length to be <= to the actual buffer size?  We know that the runtime interface has no way to validate that the buffer size matches the actual buffer.

Reply:
Are you suggesting we change the FUZZ_SIZE to be eqaul to _SIZE = len(_DATA)?

Reply:
We could set _SIZE = random.randrange(len(_DATA)) so that they may not match but we will not overflow our python buffer.